src/cpp/src/text2image/models/sd3_transformer_2d_model.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "openvino/genai/text2image/sd3_transformer_2d_model.hpp"
+
+#include "utils.hpp"
+#include "json_utils.hpp"
+#include "openvino/runtime/core.hpp"
+
+#include <fstream>
+
+namespace ov {
+namespace genai {
+
+SD3Transformer2DModel::Config::Config(const std::string& config_path) {
+    std::ifstream file(config_path);
+    OPENVINO_ASSERT(file.is_open(), "Failed to open ", config_path);
+
+    nlohmann::json data = nlohmann::json::parse(file);
+    using utils::read_json_param;
+
+    read_json_param(data, "sample_size", sample_size);
+    read_json_param(data, "patch_size", patch_size);
+    read_json_param(data, "in_channels", in_channels);
+    read_json_param(data, "num_layers", num_layers);
+    read_json_param(data, "attention_head_dim", attention_head_dim);
+    read_json_param(data, "num_attention_heads", num_attention_heads);
+    read_json_param(data, "joint_attention_dim", joint_attention_dim);
+    read_json_param(data, "caption_projection_dim", caption_projection_dim);
+    read_json_param(data, "pooled_projection_dim", pooled_projection_dim);
+    read_json_param(data, "out_channels", out_channels);
+    read_json_param(data, "pos_embed_max_size", pos_embed_max_size);
+}
+
+SD3Transformer2DModel::SD3Transformer2DModel(const std::string root_dir) :
+    m_config(root_dir + "/config.json") {
+    m_model = ov::Core().read_model(root_dir + "/openvino_model.xml");
```

Comment:
```suggestion
    m_config(root_dir + "/config.json"),
    m_model(utils::singleton_core().read_model(root_dir + "/openvino_model.xml")) {
```
There's another Core() creation below.